src/common/walFilter/greenplumCommon.h

```diff
@@ -0,0 +1,93 @@
+
+#include <stdint.h>
+#include "postgres/interface/static.vendor.h"
+
+// Common macros and constants for different versions of Greenplum
+
+// types
+typedef uint32 TimeLineID;
+typedef uint64 XLogRecPtr;
+typedef uint8 RmgrId;
+typedef uint32 pg_crc32;
+typedef uint32 BlockNumber;
+
+// constants
+/* This flag indicates a "long" page header */
+#define XLP_LONG_HEADER             0x0002
+#define MAXIMUM_ALIGNOF 8
+#define XLR_MAX_BKP_BLOCKS      4
+#define XLP_FIRST_IS_CONTRECORD     0x0001
+#define XLP_FIRST_IS_OVERWRITE_CONTRECORD 0x0008
+
+#define RESOURCE_MANAGER_XLOG_ID 0
+
+// macros
+#define TYPEALIGN(ALIGNVAL,LEN)  \
+    (((uintptr_t) (LEN) + ((ALIGNVAL) - 1)) & ~((uintptr_t) ((ALIGNVAL) - 1)))
+#define MAXALIGN(LEN)           TYPEALIGN(MAXIMUM_ALIGNOF, (LEN))
+
+#define SizeOfXLogShortPHD  MAXALIGN(sizeof(XLogPageHeaderData))
+#define SizeOfXLogLongPHD   MAXALIGN(sizeof(XLogLongPageHeaderData))
+#define XLogPageHeaderSize(hdr)     \
+    (((hdr)->xlp_info & XLP_LONG_HEADER) ? SizeOfXLogLongPHD : SizeOfXLogShortPHD)
+#define SizeOfXLogRecord    MAXALIGN(sizeof(XLogRecord))
+#define XLogRecGetData(record)  ((char*) (record) + SizeOfXLogRecord)
+#define XLR_BKP_BLOCK(iblk)     (0x08 >> (iblk))        /* iblk in 0..3 */
+#define Min(x, y)       ((x) < (y) ? (x) : (y))
+
+#define XLogSegmentsPerXLogId(segSize)   (0x100000000ULL / segSize)
+#define XLogFromFileName(fname, tli, logSegNo, segSize)  \
+    do {                                                \
+        uint32 log;                                     \
+        uint32 seg;                                     \
+        sscanf(fname, "%08X%08X%08X", tli, &log, &seg); \
+        *logSegNo = (uint64) log * XLogSegmentsPerXLogId(segSize) + seg; \
+    } while (0)
+
+// structs
+typedef struct XLogPageHeaderData
+{
+    uint16 xlp_magic;           /* magic value for correctness checks */
+    uint16 xlp_info;            /* flag bits, see below */
+    TimeLineID xlp_tli;         /* TimeLineID of first record on page */
+    XLogRecPtr xlp_pageaddr;    /* XLOG address of this page */
+
+    /*
+     * When there is not enough space on current page for whole record, we
+     * continue on the next page.  xlp_rem_len is the number of bytes
+     * remaining from a previous page.
+     *
+     * Note that xl_rem_len includes backup-block data; that is, it tracks
+     * xl_tot_len not xl_len in the initial header.  Also note that the
+     * continuation data isn't necessarily aligned.
+     */
+    uint32 xlp_rem_len;         /* total len of remaining data for record */
+} XLogPageHeaderData;
+
+typedef struct XLogLongPageHeaderData
+{
+    XLogPageHeaderData std;     /* standard header fields */
+    uint64 xlp_sysid;           /* system identifier from pg_control */
+    uint32 xlp_seg_size;        /* just as a cross-check */
+    uint32 xlp_xlog_blcksz;         /* just as a cross-check */
+} __attribute__ ((aligned (MAXIMUM_ALIGNOF))) XLogLongPageHeaderData;
```

Comment:
Still, why specify the alignment attribute here if it has already been taken into account [above](https://github.com/arenadata/pgbackrest/pull/37/files#diff-27b8ea7a0c5584c831697db920a2e88919325d6779a6019300494a9374d127e4R32)?